pkg/health/membergarbagecollector/membergarbagecollector.go

```diff
@@ -0,0 +1,171 @@
+
+	etcdclient "github.com/gardener/etcd-backup-restore/pkg/etcdutil/client"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	podNamespace = "POD_NAMESPACE"
+	podName      = "POD_NAME"
+)
+
+// MGCChecker is an interface used to unit test functions in this package
+type MGCChecker interface {
+	RemoveSuperfluousMembers(ctx context.Context) error
+}
+
+// MemberGarbageCollector contains information to regularly cleanup superfluous ETCD members
+type MemberGarbageCollector struct {
+	logger       *logrus.Entry
+	gcTimer      *time.Timer
+	k8sClient    client.Client
+	podNamespace string
+	etcdName     string
+}
+
+func init() {
+	druidv1alpha1.AddToScheme(scheme.Scheme)
+}
+
+// NewMemberGarbageCollector returns the member garbage collect object
+func NewMemberGarbageCollector(logger *logrus.Entry, clientSet client.Client) (*MemberGarbageCollector, error) {
+	if clientSet == nil {
+		return nil, &errors.EtcdError{
+			Message: "nil kubernetes clientset passed, can not create etcd member garbage collector",
+		}
+	}
+	namespace, err := utils.GetEnvVarOrError(podNamespace)
+	if err != nil {
+		logger.Fatalf("POD_NAMESPACE env var not present: %v", err)
+	}
+	podName, err := utils.GetEnvVarOrError(podName)
+	if err != nil {
+		logger.Fatalf("POD_NAME env var not present: %v", err)
+	}
+	return &MemberGarbageCollector{
+		logger:       logger.WithField("actor", "membergc"),
+		k8sClient:    clientSet,
+		podNamespace: namespace,
+		etcdName:     podName[:strings.LastIndex(podName, "-")], //TODO Evaluate if this will always be sufficient
+	}, nil
+}
+
+// RunMemberGarbageCollectorPeriodically periodically calls the member garbage collector
+func RunMemberGarbageCollectorPeriodically(ctx context.Context, hconfig *brtypes.HealthConfig, logger *logrus.Entry, etcdConfig *brtypes.EtcdConnectionConfig) {
+	clientSet, err := miscellaneous.GetKubernetesClientSetOrError()
+	if err != nil {
+		logger.Errorf("failed to create kubernetes clientset: %v", err)
+		return
+	}
+	mgc, err := NewMemberGarbageCollector(logger, clientSet)
+	if err != nil {
+		logger.Errorf("failed to initialize new etcd member garbage collector: %v", err)
+		return
+	}
+	mgc.gcTimer = time.NewTimer(hconfig.MemberGCDuration.Duration)
+	defer mgc.gcTimer.Stop()
+	mgc.logger.Info("Started etcd member garbage collector")
+
+	for {
+		//Create etcd client to get etcd cluster member list
+		clientFactory := etcdutil.NewFactory(*etcdConfig)
+		etcdCluster, err := clientFactory.NewCluster()
+		if err != nil {
+			mgc.logger.Errorf("Failed to create etcd cluster client: %v", err)
+		}
+		defer etcdCluster.Close()
+		select {
+		case <-mgc.gcTimer.C:
+			err := mgc.RemoveSuperfluousMembers(ctx, etcdCluster)
+			if err != nil {
+				mgc.logger.Warn(err)
+			}
+			mgc.gcTimer.Reset(hconfig.MemberGCDuration.Duration)
+		case <-ctx.Done():
+			mgc.logger.Info("Stopping etcd member garbage collector")
+			etcdCluster.Close()
```

Comment:
you can remove this line as above you are using `defer`  with etcdCluster.Close()